test/test_expanded_weights.py

```diff
@@ -474,6 +475,38 @@ def forward(self, input):
         expected_grads = tuple(expected_grad for expected_grad in expected_grads if expected_grad is not None)
         assert [self.assertEqual(actual, 2 * expected) for (actual, expected) in zip(actual_grads, expected_grads)]
 
+    # getting the expected per sample grads for packed sequences is non-trivial. This just smoke checks that the
+    # outputs are the same and the sum of the per sample gradients is the same as the batched gradient
```

Comment:
Why is this non-trivial? I would have expected one could convert the PackedSequence back into a list of sequences and then for each sequence, compute the gradient and finally stack all those gradients together to produce the per-sample-gradients

Reply:
Good point! I got confused with how packed sequence is implemented and forgot about the inverse operation. Added the test but the one that's a little tough now is the gradient with respect to the input since they're now taking in two different types so I left out that check